stablehlo/dialect/TypeInference.cpp

```diff
@@ -2354,6 +2354,12 @@ LogicalResult inferMapOp(
     std::optional<Location> location, ValueRange inputs,
     DenseIntElementsAttr dimensions, Region& computation,
     SmallVectorImpl<ShapedTypeComponents>& inferredReturnShapes) {
+  // map_i2
+  if (dimensions.getType().getRank() != 1)
+    return emitOptionalError(location,
+                             "dimensions should be rank 1 but got rank ",
+                             dimensions.getType().getRank());
+
   if (failed(verifyRegionNotEmpty(location, computation))) return failure();
```

Comment:
Also add a check for c2 similar to what HLO has
https://github.com/tensorflow/tensorflow/blob/4f1dd6d5123f4eb6afc85fac36df09b4a8b49c83/tensorflow/compiler/xla/service/shape_inference.cc#L1161

Reply:
The check is not needed as this is still caught by the ODS with the `verifyAtLeastNOperands(op, 1)` check.
Reference: https://github.com/llvm/llvm-project/blob/main/mlir/lib/IR/Operation.cpp#L844-L845